fastcrypto/src/groups/mod.rs

```diff
@@ -44,3 +44,14 @@ pub trait Scalar:
 {
     fn rand<R: AllowedRng>(rng: &mut R) -> Self;
 }
+
+pub trait Pair {
+    type Other;
+    type Output;
+
+    fn pair(&self, other: &Self::Other) -> Self::Output;
+}
+
+pub trait FromHashedMessage {
+    fn hash(msg: &[u8]) -> Self;
```

Comment:
should this takes in a SHA3_256 type instead of raw bytes? https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-06#section-8.7.1

Reply:
It does the hashing using the hash function defined by the RFC.
What indeed we may do here is to receive also the domain separator (DST in our code) and pass it to blst. Currently I'm using the standard ones since I don't see why we would want to use, e.g., PoP here. If we need that in the future, we can add the DST string as a parameter.